src/PhenX.EntityFrameworkCore.BulkInsert/BulkInsertProviderBase.cs

```diff
@@ -148,11 +149,16 @@ public virtual async Task<List<T>> BulkInsertReturnEntities<T>(
         CancellationToken ctk = default
     ) where T : class
     {
+        if (options is not TOptions providerOptions)
```

Comment:
I would still allow the base options

Reply:
Base options are still allowed with the current syntax

Reply:
But BaseOption is not a TOption...

Reply:
The `options` instance is created in the provider specific GetDefaultOptions methode, which returns a provider specific class (or the base class if the provider does not have specific options)

Reply:
Makes sense.